src/mon.c

```diff
@@ -2928,9 +2928,12 @@ xkilled(
             u.ublessed = 0;
         if (mdat->maligntyp == A_NONE)
             adjalign((int) (ALIGNLIM / 4)); /* BIG bonus */
-    } else if (mtmp->mtame) {
+    } else if ((mtmp->mtame) 
+               && (wep && wep->oartifact != ART_STORMBRINGER)
+               && (u.ualign.type != CHAOTIC)) {
```

Comment:
Isn't the logic backwards here? `mtmp->mtame && !(wep && wep->oartifact == ART_STORMBRINGER && u.ualign.type == CHAOTIC` seems to be the intended condition.